openbas-api/src/test/java/io/openbas/rest/exercise/ExerciseApiStatusTest.java

```diff
@@ -176,28 +184,49 @@ void manualStartExerciseTest() throws Exception {
     // -- PREPARE--
     ExerciseUpdateStatusInput input = new ExerciseUpdateStatusInput();
     input.setStatus(ExerciseStatus.RUNNING);
-
-    // -- EXECUTE --
-    String response =
-        mvc.perform(
-                put(EXERCISE_URI + "/" + SCHEDULED_EXERCISE.getId() + "/status")
-                    .content(asJsonString(input))
-                    .contentType(MediaType.APPLICATION_JSON)
-                    .accept(MediaType.APPLICATION_JSON))
-            .andExpect(status().is2xxSuccessful())
-            .andReturn()
-            .getResponse()
-            .getContentAsString();
-
-    // -- ASSERT --
-    Thread.sleep(2000);
-    List<ExecutableInject> injects = injectHelper.getInjectsToRun();
-    Instant nextMinute = now().truncatedTo(MINUTES).plus(1, MINUTES);
-    assertEquals(nextMinute.toString(), JsonPath.read(response, "$.exercise_start_date"));
-    assertEquals(
-        Arrays.asList(ExerciseStatus.CANCELED.name(), ExerciseStatus.PAUSED.name()),
-        JsonPath.read(response, "$.exercise_next_possible_status"));
-    assertEquals(1, injects.size());
+    Instant instantBeforeSetRun = REFERENCE_TIME;
+    Instant instantAfterSetRun = REFERENCE_TIME.plus(1, MINUTES);
```

Comment:
question: what's the average time the test takes to run ? Will 1 minute always be enough ?

Reply:
It's not a duration, they are two different points in time to simulate a real-world scenario when we check if an inject is ready to run: "Now" must be later or equal to "startTime". Here I'm forcing it to be slightly later.